Add a guide for reproducing scripts for Bundler (#3426)
rubygems/rubygems: bundler/doc/contributing/REPRODUCTION_SCRIPT_GUIDE.md

# Reproduction Script Guide

Having a reproduction script of an issue helps us a lot with processing it faster. We know some of our contributors
could not be familiarized with building reproduction scripts, so in this guide, we'll help you understand what is a 
reproduction script and provide a template and a couple of examples of some reproduction scripts.

> ## What is a reproduction script?

Reviewer: ```suggestion
Having a reproduction script of an issue helps us a lot with processing it faster. We know some of our contributors
may not be familiar with building reproduction scripts, so in this guide, we'll help you understand what a 
reproduction script is and provide a template and a couple of examples of some reproduction scripts.

## What is a reproduction script?
```

A reproduction script is basically a script that reproduces an event. Usually, they are used to reproduce issues, bugs, 

> errors and/or warnings.

Reviewer: ```suggestion
A reproduction script **reproduces an event**, e.g. an issue, a bug, an error or a warning.
```

> The core idea behind creating a reproduction script is to configure the different environment variables to be as similar as

Reviewer: Perhaps the wording "the different environment variables" could be "the environment".

(Other things than ENV can act on the execution environment of the repro script. We want to avoid confusion, here.)

Author: Solved in commit abdeade.

Thanks for the suggestion.

possible with the environment trying to be reproduced.

After the environment has been configured properly the idea is to reproduce a series of events until reaching a specific 
point, which as mentioned before could be a bug, error, warning, or any other event, by using a series of commands.

## A template for reproduction scripts

> Below you can find a template that may be used in order to create reproduction scripts related to bundle issues:

Reviewer: ```suggestion
This template can be used to create reproduction scripts related to Bundler issues:
```

```sh
# Add here the code that could be used as a template for reproduction scripts.

# Be sure to include rich helpful comments which could help the readers understand the logic 
# behind the process.
```

## Reproduction scripts examples

> Below you can find a couple of examples of reproduction scripts:

Reviewer: ```suggestion
Examples of reproduction scripts:

```

Reviewer: Or skip the introductory sentence competely.
